### PR TITLE
fby3.5: cl:Add I2C slave relative file

### PR DESCRIPTION
**Pull request by MouchenHung-QUANTA**
Summary:
- Provide I2C slave callback functions and user APIs.
- In order to let I2C slave could be registered optionally while runtime.

Test plan:
- Build code: Pass
- User APIs validation: Pass
- Stress test: Pass

### CODE REVIEW

File: common/util/hal_i2c_slave.c [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+  if (data->buffer_idx){
+    data->current_msg.msg_length = data->buffer_idx;
+
+#if DEBUG_MODE
+    printk("recv: ");
+    for (int i=0; i<data->current_msg.msg_length; i++){
+      printk("[%d]", data->current_msg.msg[i]);
+    }
+    printk("\n");
+#endif
```
Comment (GoldenBug) [inline common/util/hal_i2c_slave.c:112]: Is this debug code necessary or just left here from testing?
If not please remove it.

**Review by GoldenBug — CHANGES_REQUESTED** [review]
Please make changes and update pull request, don't make a new one.

File: common/util/hal_i2c_slave.c (hunk 1 above)
Comment (MouchenHung) [reply common/util/hal_i2c_slave.c:112]: Thanks for code review, it's for debug-use to check whether i2c slave ISR function received data correctly. We will remove it later update.

Comment (MouchenHung-QUANTA) [reply common/util/hal_i2c_slave.c:112]: Thanks for code review, it's for debug-use to check whether i2c slave ISR function received data correctly. We will remove it later update.

**Pull request closed without merging**